### ISSUE TITLE
Training from scratch results in high FID

### ISSUE THREAD

**Issue by hu-jy-0406** [LTH14/mar#105]
Thanks for your great work!
I tried training MAR-B from scratch on ImageNet-1k using the following script:
```
set -x

export CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7
export PYTHONPATH=***:$PYTHONPATH
export WANDB_API_KEY=***

torchrun \
--nproc_per_node=8 --nnodes=1 --node_rank=0 \
--master_addr=localhost \
--master_port=11234 \
mar/main_mar.py \
--img_size 256 \
--vae_path ***kl16.ckpt \
--vae_embed_dim 16 \
--vae_stride 16 \
--patch_size 1 \
--model mar_base \
--diffloss_d 6 \
--diffloss_w 1024 \
--num_iter 64 \
--num_sampling_steps 100 \
--cfg 2.9 \
--cfg_schedule linear \
--temperature 1.0 \
--epochs 400 \
--warmup_epochs 100 \
--batch_size 128 \
--blr 1.0e-4 \
--diffusion_batch_mul 4 \
--output_dir *** \
--resume *** \
--data_path ***\
--online_eval \
--log_dir *** \
--eval_freq 50 \
--save_last_freq 5 \
--num_images 50000 \
--class_num 1000 \
--eval_bsz 256 \
--ema_rate 0.9995 \
--use_cached \
--cached_path *** \
--num_workers 24 \
--pin_mem
```
After 150 epochs, the FID reached 85.67, but it increased to 96.95 at 200 epochs, so I stopped the training. The training curves and generated images are attached below:

<img width="2988" height="649" alt="Image" src="https://github.com/user-attachments/assets/d5ea9aed-ab33-47cc-a3bc-fa628634fd8b" />

<img width="380" height="241" alt="Image" src="https://github.com/user-attachments/assets/faf916a0-55b4-4566-9538-c57b469acdfe" />

<img width="380" height="241" alt="Image" src="https://github.com/user-attachments/assets/cee56d6c-f230-4949-b096-80c53c59e7f9" />

The training was done on 8×A100 GPUs. I was able to reproduce the reported FID 2.31 using your pretrained MAR-B checkpoint, but I failed to reproduce it when training from scratch.

Do you know what might have caused this issue? Could it be due to my larger batch size or the different EMA rate? Also, could you please share your training loss curve and FID curve for MAR-B?

Thanks a lot for your kind help!

**Comment by LTH14** [comment]
Thanks for your interest. The generated images look quite reasonable, and from my experience, this should not be images from FID ~90. So I would suggest you take a look at your evaluation script -- maybe just uncomment here https://github.com/LTH14/mar/blob/main/engine_mar.py#L218 and check the 50000 images you generated.

**Comment by hu-jy-0406** [comment]
Thanks a lot for your reply! I rewrote my code and retrained MAR-B from scratch again using the default settings. Since training for 800 epochs is very time-consuming, I only trained for 400 epochs. This time I obtained the following FID curve with cfg=1.0. Does this look correct?
<img width="2315" height="317" alt="Image" src="https://github.com/user-attachments/assets/b137a1c2-3437-49b1-9fc8-7acce8f12b51" />

**Comment by LTH14** [comment]
Yes this is correct

**Comment by hu-jy-0406** [comment]
Okay~Thank you!
